File: app/src/main/java/com/babestudios/companyinfouk/ui/officers/OfficersAdapter.java
package com.babestudios.companyinfouk.ui.officers;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.babestudios.companyinfouk.R;
import com.babestudios.companyinfouk.data.model.officers.OfficerItem;
import com.babestudios.companyinfouk.data.model.officers.Officers;

import butterknife.BindView;
import butterknife.ButterKnife;

public class OfficersAdapter extends RecyclerView.Adapter<OfficersAdapter.OfficersViewHolder> {

	private OfficersRecyclerViewClickListener mItemListener;

	private Officers officers = new Officers();

	OfficersAdapter(Context c, Officers officers) {
		mItemListener = (OfficersRecyclerViewClickListener) c;
		this.officers = officers;
	}

	@Override
	public OfficersViewHolder onCreateViewHolder(ViewGroup parent, int i) {
		View itemLayoutView = LayoutInflater.from(parent.getContext())
				.inflate(R.layout.officers_list_item, parent, false);

		return new OfficersViewHolder(itemLayoutView);
	}

	@Override
	public void onBindViewHolder(OfficersViewHolder viewHolder, int position) {

		viewHolder.lblName.setText(officers.items.get(position).name);
		viewHolder.lblAppointedOn.setText(officers.items.get(position).appointedOn);
		viewHolder.lblRole.setText(officers.items.get(position).officerRole);
		viewHolder.lblResignedOn.setText(officers.items.get(position).resignedOn);

	}

	public long getItemId(int position) {
		return position;
	}

	@Override
	public int getItemCount() {
		return officers.items.size();
	}

	class OfficersViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
		@BindView(R.id.lblName)
		TextView lblName;
		@BindView(R.id.lblAppointedOn)
		TextView lblAppointedOn;
		@BindView(R.id.lblRole)
		TextView lblRole;
		@BindView(R.id.lblResignedOn)
		TextView lblResignedOn;

		OfficersViewHolder(View itemView) {
			super(itemView);
			ButterKnife.bind(this, itemView);
			itemView.setOnClickListener(this);
		}

		@Override
		public void onClick(View v) {
			mItemListener.officersItemClicked(v, this.getLayoutPosition(), officers.items.get(getLayoutPosition()));
		}
	}

	interface OfficersRecyclerViewClickListener {
		void officersItemClicked(View v, int position, OfficerItem officerItem);
	}

	void updateItems(Officers officers) {
		this.officers = officers;
		notifyDataSetChanged();
	}
}

File: app/src/main/java/com/babestudios/companyinfouk/ui/officers/OfficersAdapter.kt
package com.babestudios.companyinfouk.ui.officers

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.babestudios.companyinfouk.R
import com.babestudios.companyinfouk.data.model.officers.OfficerItem
import com.babestudios.companyinfouk.data.model.officers.Officers

import butterknife.BindView
import butterknife.ButterKnife

class OfficersAdapter internal constructor(c: Context, officers: Officers) : RecyclerView.Adapter<OfficersAdapter.OfficersViewHolder>() {

	private val mItemListener: OfficersRecyclerViewClickListener

	private var officers = Officers()

	init {
		mItemListener = c as OfficersRecyclerViewClickListener
		this.officers = officers
	}

	override fun onCreateViewHolder(parent: ViewGroup, i: Int): OfficersViewHolder {
		val itemLayoutView = LayoutInflater.from(parent.context)
				.inflate(R.layout.row_officers, parent, false)

		return OfficersViewHolder(itemLayoutView)
	}

	override fun onBindViewHolder(viewHolder: OfficersViewHolder, position: Int) {

		viewHolder.lblName?.text = officers.items[position].name
		viewHolder.lblAppointedOn?.text = officers.items[position].appointedOn
		viewHolder.lblRole?.text = officers.items[position].officerRole
		viewHolder.lblResignedOn?.text = officers.items[position].resignedOn

	}

	override fun getItemId(position: Int): Long {
		return position.toLong()
	}

	override fun getItemCount(): Int {
		return officers.items.size
	}

	inner class OfficersViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
		@JvmField
		@BindView(R.id.lblName)
		var lblName: TextView? = null
		@JvmField
		@BindView(R.id.lblAppointedOn)
		var lblAppointedOn: TextView? = null
		@JvmField
		@BindView(R.id.lblRole)
		var lblRole: TextView? = null
		@JvmField
		@BindView(R.id.lblResignedOn)
		var lblResignedOn: TextView? = null

		init {
			ButterKnife.bind(this, itemView)
			itemView.setOnClickListener(this)
		}

		override fun onClick(v: View) {
			mItemListener.officersItemClicked(v, this.layoutPosition, officers.items[layoutPosition])
		}
	}

	internal interface OfficersRecyclerViewClickListener {
		fun officersItemClicked(v: View, position: Int, officerItem: OfficerItem)
	}

	internal fun updateItems(officers: Officers) {
		this.officers = officers
		notifyDataSetChanged()
	}
}
